drush.php

```diff
@@ -54,6 +55,14 @@
     throw new \Exception("Could not locate autoload.php. cwd is $cwd; __DIR__ is " . __DIR__);
 }
 
+if(array_key_exists('args', $_SERVER)) {
+    $argv = explode(' ',$_SERVER['args']);
```

Comment:
I think we'd need to do a better job here if we're going to support this, as a simple `explode` on spaces provides no way for quoted / escaped arguments.  I wouldn't expect that all of the tests would pass until this is right.

Reply:
I've tried something with regex.
But not sure, must be tested with real quoted param

Reply:
another way should be to encode each base 64 param in a shell script calling drush with php-fpm and keep space as separator for arg varaiable and add a the base 64 decoding after the explode

What do you think the best way ?

Reply:
I think we need to take a step back and figure out what the use-case is, and how that would relate to the standard Drush that uses php-cli.

If you are just making your own fork of Drush, then you can base64 or json encode your arguments and do whatever you want, and it really won't matter. I don't have a strong opinion on what you do in this case; it's whatever is most convenient for you.

This is looking a little too edge-case for inclusion in Drush core at the moment, though. The regex parser is fragile and doesn't give good error messages for unbalanced quotes, et. al.; the base64 method would be inconvenient for most clients. There is also the question of selecting between ConsoleOutput and an OutputBuffer. It is unclear on how this would be installed an used.

So I'd try to define those items first. Start off assuming this will be your own thing. Make your own copy of drush.php or whatever to get you going. Then figure out what your cli client is going to look like, and how it's going to identify and call your php-fpm-embedded Drush. Ultimately, this could probably become something that one composer-require's inside of a Drupal project along with drush/drush:^9.

Reply:
ok got something with encoding (url one, will need to install the urlencode shell command).
Use case is case (for me) running drush migrate import by cron every 15 minutes for  importing remote content (one from youtube, other via rss feed of a few websites).
But everyone  using drush on production server may be interested by taking profit of opcache as drupal class are in cache.
For consoleOutput i have an  idea, will send a patch soon that define a child class of consoleOutput and overriding the constructor to open php://stderr and php://output insteand of  php://stderr and php://stdout